### PR TITLE
DCS-1196 new prison/temporary-absences/{agencyId}   endpoint

### PR DESCRIPTION
**Pull request by GurnankCheema**
endpoint return stubbed data for temporary absences

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/PrisonResource.kt [hunk 1]
```diff
@@ -70,4 +70,10 @@ class PrisonResource(
     @Schema(description = "Offender Number", example = "A12345", required = true)
     @PathVariable offenderNumber: String
   ) = prisonService.getPrisonerImage(offenderNumber)
+
+  @GetMapping(value = ["/temporary-absences/{agencyId}"], produces = [MediaType.APPLICATION_JSON_VALUE])
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/PrisonResource.kt:74]: will need to add the response annotations as above for the swagger doc

Comment (GurnankCheema) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/PrisonResource.kt]: done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.welcometoprison.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.welcometoprison.model.TemporaryAbsence
+import uk.gov.justice.digital.hmpps.welcometoprison.model.prison.PrisonService
+
+@RestController
+@Validated
+@PreAuthorize("hasRole('SYSTEM_USER')")
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt:22]: Can we use: `ROLE_VIEW_INCOMING_MOVEMENTS` ??

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+@RequestMapping(name = "Prison", path = ["/temporary-absences"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class TemporaryAbsencesResource(
+  private val prisonService: PrisonService,
+) {
+  @Operation(
+    summary = "Retrieves the temporary absences for a prison",
+    description = "Retrieves the current temporary absences information, role required is SYSTEM_USER",
+    security = [SecurityRequirement(name = "SYSTEM_USER", scopes = ["read"])],
```
Comment (GurnankCheema) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt:30]: Not absolutely sure whether access should be restricted the SYSTEM_USER role or indeed any role at all.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt (hunk 2 above)
Comment (GurnankCheema) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt]: yes

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**